Sourcery/Models/Method.swift

```diff
@@ -13,32 +13,21 @@ class Method: NSObject, AutoDiffable {
         let name: String
 
         /// Parameter type name
-        let typeName: String
+        let typeName: TypeName
```

Comment:
How about adding a common protocol for `Variable`, `Method.Parameter`, `AssociatedValue` like this:

```swift
protocol Typed {
  var type: Type? { get set }
  var typeName: TypeName { get }
  var isOptional: Bool { get }
  var unwrappedTypeName: String { get }
}
extension Typed {
  var isOptional: Bool { return typeName.isOptional }
  var unwrappedTypeName: String { return typeName.unwrappedTypeName }
}
```